src/metabase/models/params.clj

```diff
@@ -28,8 +28,9 @@
     [:field field-form nil]
     (mbql.u/match-one field-form :field)))
 
+;; TODO -- I'm not 100% sure we need this code anymore.
```

Comment:
It gets used in `metabase.driver.common.parameters.dates/date-string->filter` which in turn gets used in `metabase.models.params.chain-filter` with this `field-id` semantics, and a few other places like that.

Reply:
True. I'll remove this comment